### PR TITLE
fix: Add back missing Any in and Not in operators for Single Select 

### PR DESCRIPTION
**Pull request by hariombalhara**
## What does this PR do?

Fixes single select field missing "Any in" and "Not in" operators.

As reported on intercom @joseuo94 

Regression from #16823

[Testing loom for operator fix](https://www.loom.com/share/e0cd7dc263e74c98882a34bceff329bf)
[Loom for new queryValue validation](https://www.loom.com/share/3ee395167dce4bd3a49b3892d7b89bd4)

image
image

## Before:
image

## After:
Multiselect still doesn't support "Any in" and "Not any in" as intended for form fields query builder.
image

Single Select is back supporting "Any in" and "Not any in"
image

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] N/A I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). If N/A, write N/A here and check the checkbox.
- [x] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?
See loom

## Ensuring similar regression doesn't occur again
- Added tests that ensure that RAQB config is correct
- During save the queryValue is now validated in a stricter fashion for missing value. Missing value happens if RAQB config is messed up. It allows wrong values to be not saved and thus allows Routing Forms to recover automatically once we fix the issue.

### CODE REVIEW

File: packages/app-store/routing-forms/components/react-awesome-query-builder/config/BasicConfig.ts [hunk 1]
```diff
@@ -356,8 +356,8 @@ const types: Types = {
       },
       multiselect: {
         operators: [
-          //   "select_any_in",
-          //   "select_not_any_in",
+          "select_any_in",
+          "select_not_any_in",
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/components/react-awesome-query-builder/config/BasicConfig.ts:360]: I accidentally commented these thinking that I was removing them for multiselect field but it is actually a case of `Select` field which is suppsoed to have `any_in` and `not_any_in` operators that use multiselect widget for operand.

File: packages/app-store/routing-forms/__tests__/config.test.ts [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+  expect(config.types.select.widgets.multiselect.operators).not.toContain("select_not_equals");
+
+
+  expect(config.types.select.widgets.select.operators).toContain("select_equals");
+  expect(config.types.select.widgets.select.operators).toContain("select_not_equals");
+
+  // Important to verify that select_any_in and select_not_any_in are not present in select as that might cause select widget operand to show up for these operators
+  expect(config.types.select.widgets.select.operators).not.toContain("select_any_in");
+  expect(config.types.select.widgets.select.operators).not.toContain("select_not_any_in");
+};
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/__tests__/config.test.ts:52]: It verifies that config isn't broken and avoids regression in future.

File: packages/app-store/routing-forms/components/react-awesome-query-builder/config/BasicConfig.ts [hunk 3]
```diff
@@ -343,10 +343,6 @@ const types: Types = {
         operators: [
           "select_equals",
           "select_not_equals",
-          // "is_empty",
-          // "is_not_empty",
-          "is_null",
-          "is_not_null",
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/components/react-awesome-query-builder/config/BasicConfig.ts:349]: is_null and is_not_null operators aren't defined. So, it wasn't used.

File: packages/app-store/routing-forms/zod.ts [hunk 4]
```diff
@@ -55,6 +55,64 @@ const queryValueSchema = z.object({
...
+            typeof value === "object" && value !== null;
+          Object.entries(children1).forEach(([, _rule]) => {
+            const rule = _rule as unknown;
+            if (!isObject(rule) || rule.type !== "rule") return;
+            if (!isObject(rule.properties)) return;
+
+            const value = rule.properties.value || [];
+            if (!(value instanceof Array)) {
+              return;
+            }
```
Comment (hariombalhara) [inline packages/app-store/routing-forms/zod.ts:92]: Be lenient with what is passed by RAQB and just focus on one thing which is the value.

**Review by CarinaWolli — APPROVED** [review]

**Pull request merged**